chore/document-endpoints

plysrh opened the pull request:
## Summary
Consume data from related endpoints

## Changes
- [x] Upload images and documents when creating a new product
- [x] Improve syntax
- [x] Consuming get spec blocks endpoint
- [x] Consuming add spec block endpoint
- [x] Consuming add spec block text endpoint
- [x] Update specs page path
- [x] Add services to create, update and remove blocks
- [x] Add services to create, update and remove blocks images
- [x] Add services to create, update and remove blocks texts
- [x] Sort blocks
- [x] Edit and delete image
- [x] Improve blocks's text format
- [x] Improve spec products list behavior and dimentions

## Tickets
- https://trello.com/c/9q2vO0cG
- https://trello.com/c/m5tgrOu2
- https://trello.com/c/2gB90FLA
- https://trello.com/c/lGTGyl8X
- https://trello.com/c/QIVRMOYF
- https://trello.com/c/rHnAAaMt
- https://trello.com/c/RxJJuwqs
- https://trello.com/c/XqnScOxq
- https://trello.com/c/9p2deBmm

j0araya commented:
Indice al producto agregado en el documento. [Trello](https://trello.com/c/m5tgrOu2/15-hito-12-mostrar-lista-de-bloques-de-la-especificaci%C3%B3n)

Error al crear producto 
![image](https://user-images.githubusercontent.com/11578634/88127664-402c3180-cba2-11ea-9446-5101ae94c52c.png)

Payload
![image](https://user-images.githubusercontent.com/11578634/88127755-78cc0b00-cba2-11ea-98d8-99f3af7623fb.png)


Buscador: Apretar "escape" para salir del buscador de productos no funciona, en el modal si.[Trello](https://trello.com/c/9p2deBmm/4-editor-de-especificaciones-modal-infinito-al-buscar-productos)

Agregar un mismo producto dos veces hace fallar el documento (drag and drop)
.

plysrh commented:
> Indice al producto agregado en el documento. [Trello](https://trello.com/c/m5tgrOu2/15-hito-12-mostrar-lista-de-bloques-de-la-especificaci%C3%B3n)
> 
> Error al crear producto
> ![image](https://user-images.githubusercontent.com/11578634/88127664-402c3180-cba2-11ea-9446-5101ae94c52c.png)
> 
> Payload
> ![image](https://user-images.githubusercontent.com/11578634/88127755-78cc0b00-cba2-11ea-98d8-99f3af7623fb.png)
> 
> Buscador: Apretar "escape" para salir del buscador de productos no funciona, en el modal si.[Trello](https://trello.com/c/9p2deBmm/4-editor-de-especificaciones-modal-infinito-al-buscar-productos)
> 
> Agregar un mismo producto dos veces hace fallar el documento (drag and drop)
> .

1.- El índice no sé cómo va a manejar, lo veo difícil por el lado del front-end
2.- Pude encontrar otro error que tiene que ver con el endpoint para subir documentos, pero no logro reproducir que muestras, ¿podrías intentar reproducirlo nuevamente y especificar los pasos que seguiste? 
3.- Fixed
4.- Fixed
